x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/FieldNameTranslators.java

```diff
@@ -35,75 +36,42 @@
-    static final List<FieldNameTranslator> FIELD_NAME_TRANSLATORS;
-
-    static {
-        FIELD_NAME_TRANSLATORS = List.of(
-            new ExactFieldNameTranslator(s -> "creator.principal", "username"),
-            new ExactFieldNameTranslator(s -> "creator.realm", "realm_name"),
-            new ExactFieldNameTranslator(s -> "name", "name"),
-            new ExactFieldNameTranslator(s -> API_KEY_TYPE_RUNTIME_MAPPING_FIELD, "type"),
-            new ExactFieldNameTranslator(s -> "creation_time", "creation"),
-            new ExactFieldNameTranslator(s -> "expiration_time", "expiration"),
-            new ExactFieldNameTranslator(s -> "api_key_invalidated", "invalidated"),
-            new ExactFieldNameTranslator(s -> "invalidation_time", "invalidation"),
-            // allows querying on all metadata values as keywords because "metadata_flattened" is a flattened field type
-            new ExactFieldNameTranslator(s -> "metadata_flattened", "metadata"),
-            new PrefixFieldNameTranslator(s -> "metadata_flattened." + s.substring("metadata.".length()), "metadata.")
-        );
-    }
+public final class FieldNameTranslators {
 
-    /**
-     * Adds the {@param fieldSortBuilders} to the {@param searchSourceBuilder}, translating the field names,
-     * form query level to index level, see {@link #translate}.
-     * The optional {@param visitor} can be used to collect all the translated field names.
-     */
-    public static void translateFieldSortBuilders(
-        List<FieldSortBuilder> fieldSortBuilders,
-        SearchSourceBuilder searchSourceBuilder,
-        @Nullable Consumer<String> visitor
-    ) {
-        final Consumer<String> fieldNameVisitor = visitor != null ? visitor : ignored -> {};
-        fieldSortBuilders.forEach(fieldSortBuilder -> {
-            if (fieldSortBuilder.getNestedSort() != null) {
-                throw new IllegalArgumentException("nested sorting is not supported for API Key query");
-            }
-            if (FieldSortBuilder.DOC_FIELD_NAME.equals(fieldSortBuilder.getFieldName())) {
-                searchSourceBuilder.sort(fieldSortBuilder);
-            } else {
-                final String translatedFieldName = translate(fieldSortBuilder.getFieldName());
-                fieldNameVisitor.accept(translatedFieldName);
-                if (translatedFieldName.equals(fieldSortBuilder.getFieldName())) {
-                    searchSourceBuilder.sort(fieldSortBuilder);
-                } else {
-                    final FieldSortBuilder translatedFieldSortBuilder = new FieldSortBuilder(translatedFieldName).order(
-                        fieldSortBuilder.order()
-                    )
-                        .missing(fieldSortBuilder.missing())
-                        .unmappedType(fieldSortBuilder.unmappedType())
-                        .setFormat(fieldSortBuilder.getFormat());
+    public static final FieldNameTranslators API_KEY_FIELD_NAME_TRANSLATORS = new FieldNameTranslators(
```

Comment:
This is optional, but it feels like these belong somewhere API key specific (just like the user ones below and the roles when we add them). Since they're reused, it might be hard to find a good home for them. Just wanted to point it out if you can think of something.

Reply:
I was thinking more like someplace .security index mapping specific...
They're not too bad here, I think:
It's a place where all field name translation rules are defined and can be extended or built upon, and both index-level and query-level field names are very rarely changed (I mean there's no other place where they'd change together with something else).